partner/WHMCS/modules/servers/jetpack/jetpack.php

```diff
@@ -0,0 +1,288 @@
+<?php
+
+use WHMCS\Database\Capsule;
+
+
+if (!defined("WHMCS")) {
+    die("This file cannot be accessed directly");
+}
+
+
+/**
+ * A WHMCS module for use by Jetpack hosting partners to provision Jetpack plans.
+ * The module provides functionality for partner hosts to be able to save their
+ * client id and secret to request an access token for provisioning plans.
+ *
+ * Plans available for provisioning include free, personal, premium and professional
+ *
+ * A host has options to either provision(Create == WHMCS equivalent functional term)
+ * or Cancel(Terminate == WHMCS equivalent functional term) from the WHMCS client area.
+ *
+ * Host setup for custom fields is currently required in order to use the module.
+ *
+ */
+
+/**
+ * Jetpack Meta Data for WHMCS module.
+ * @return array
+ */
+function jetpack_MetaData()
+{
+    return array(
+        'DisplayName' => 'Jetpack by Automattic',
+        'Description' => 'Use this module to provision Jetpack plans with your Jetpack hosting partner account',
+        'APIVersion' => '1.1',
+        'RequiresServer' => false,
+    );
+}
+
+
+/**
+ * Basic configuration options required for a partner to get
+ * a Jetpack plan provisioned. Currently a partner client id
+ * and secret are the only host partner options needed to get
+ * an access token to provision a Jetpack plan
+ * @return array
+ */
+function jetpack_ConfigOptions()
+{
+    return array(
+        'Jetpack Partner Client ID' => array(
+            'Type' => 'text',
+            'Size' => '256',
+        ),
+        'Jetpack Partner Client Secret' => array(
+            'Type' => 'text',
+            'Size' => '256',
+        )
+    );
+}
+
+
+/**
+ * Equivalent to /provision. Create a Jetpack plan using
+ * a Jetpack Hosting partner account.
+ *
+ *
+ * @param array $params
+ * @return string 'success'
+ * @throws Exception
```

Comment:
It doesn't look like the function is throwing `Exception` and also it can return strings different than “success”.

Reply:
@nb I've updated the doc blocks for this function to indicate that 'success' is only expected by WHMCS when the process is complete otherwise an error string should be returned. Both the jetpack_createAccount and jetpack_terminateAccount functions will use the make_api_request function which will throw an exception if there is a curl error or the response body from the request is empty. I've refactored make_api_request a bit also and will post some updates in the P2 about this